solidity/contracts/Coordinator.sol

```diff
@@ -3,10 +3,36 @@ pragma solidity ^0.4.24;
 // Coordinator handles oracle service aggreements between one or more oracles.
 contract Coordinator {
 
+  struct ServiceAgreement {
+    uint256 payment;
+    uint256 expiration;
+    bytes32 jobSpecID;
```

Comment:
Can we rename this something like `requestDigest`? The jobSpecID is random and not shared between nodes, but the request digest should be deterministic and shared.